src/sentry/static/sentry/app/components/projectSelector.jsx

```diff
@@ -0,0 +1,165 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import {sortArray} from 'app/utils';
+import {t} from 'app/locale';
+import Button from 'app/components/button';
+import DropdownAutoComplete from 'app/components/dropdownAutoComplete';
+import Highlight from 'app/components/highlight';
+import IdBadge from 'app/components/idBadge';
+import space from 'app/styles/space';
+
+class ProjectSelector extends React.Component {
+  static propTypes = {
+    // Accepts a project id (slug) and not a project *object* because ProjectSelector
+    // is created from Django templates, and only organization is serialized
+    projectId: PropTypes.string,
+    organization: PropTypes.object.isRequired,
+
+    // Callback when a project is selected
+    onSelect: PropTypes.func,
+  };
+
+  static defaultProps = {
+    projectId: null,
+    onSelect: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeProject: this.getActiveProject(),
+    };
+  }
+
+  urlPrefix() {
+    return `/organizations/${this.props.organization.slug}`;
+  }
+
+  getActiveProject() {
+    const {projectId} = this.props;
+    return this.getProjects().find(({slug}) => slug === projectId);
+  }
+
+  getProjects() {
+    const {organization} = this.props;
+    return organization.projects.filter(project => project.isMember);
+  }
+
+  getProjectLabel(project) {
+    return project.slug;
+  }
```

Comment:
Something @mnoble recently turned me onto was using [`getter`s](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get). Have you considered starting to use these?

Reply:
It's not an option here since this takes an argument, but in general no, I think it's better to remain consistent with the rest of the codebase otherwise it'd lead to confusion

Reply:
Is this function even really needed? Seems a little unnecessary if it's just accessing a property.

Reply:
Ya probably not, it was just existing code.